tests/scipy_signal_test.py

```diff
@@ -376,6 +383,59 @@ def testWelchWithDefaultStepArgsAgainstNumpy(
     self._CheckAgainstNumpy(osp_fun, jsp_fun, args_maker, rtol=tol, atol=tol)
     self._CompileAndCheck(jsp_fun, args_maker, rtol=tol, atol=tol)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name":
+          f"_shape={jtu.format_shape_dtype_string(shape, dtype)}"
+          f"_fs={fs}_window={window}_boundary={boundary}"
+          f"_nperseg={nperseg}_noverlap={noverlap}_onesided={onesided}"
+          f"_timeaxis={timeaxis}_freqaxis{freqaxis}_nfft={nfft}",
+       "shape": shape, "dtype": dtype, "fs": fs, "window": window,
+       "nperseg": nperseg, "noverlap": noverlap, "nfft": nfft,
+       "onesided": onesided, "boundary": boundary,
+       "timeaxis": timeaxis, "freqaxis": freqaxis}
+      for shape, nperseg, noverlap, timeaxis, freqaxis in istft_test_shapes
+      for dtype in default_dtypes
+      for fs in [1.0, 16000.0]
+      for window in ['boxcar', 'triang', 'blackman', 'hamming', 'hann']
+      for nfft in [None, nperseg, int(nperseg * 1.5), nperseg * 2]
+      for onesided in [False, True]
+      for boundary in [False, True]))
+  @jtu.skip_on_devices("rocm")  # will be fixed in ROCm 5.1
+  def testIstftAgainstNumpy(self, *, shape, dtype, fs, window, nperseg,
+                            noverlap, nfft, onesided, boundary,
+                            timeaxis, freqaxis):
+    if not onesided:
+      new_freq_len = (shape[freqaxis] - 1) * 2
+      shape = shape[:freqaxis] + (new_freq_len ,) + shape[freqaxis + 1:]
+
+    def osp_fun(x):
+      # Ignore UserWarning in osp so we can also test over ill-posed cases.
+      with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        result = osp_signal.istft(
+            x,
+            fs=fs, window=window, nperseg=nperseg, noverlap=noverlap,
+            nfft=nfft, input_onesided=onesided, boundary=boundary,
+            time_axis=timeaxis, freq_axis=freqaxis)
+      return result
+
+    jsp_fun = partial(jsp_signal.istft,
+        fs=fs, window=window, nperseg=nperseg, noverlap=noverlap, nfft=nfft,
+        input_onesided=onesided, boundary=boundary,
+        time_axis=timeaxis, freq_axis=freqaxis)
```

Comment:
You might consider making `fs` one of the arguments to `osp_fun`/`jsp_fun` so that we test dynamic inputs for that value. If you remove the `float` cast in the implementation, I believe it will work.

Reply:
Done.